otx/algorithms/segmentation/adapters/mmseg/task.py

```diff
@@ -0,0 +1,563 @@
+                if backend_config.get("mo_options") is None:
+                    backend_config.mo_options = ConfigDict()
+                mo_options = backend_config.mo_options
+                if mo_options.get("args") is None:
+                    mo_options.args = ConfigDict()
+                if mo_options.get("flags") is None:
+                    mo_options.flags = []
+
+                # already defiend options have higher priority
+                options["args"].update(mo_options.args)
+                mo_options.args = ConfigDict(options["args"])
+                # make sure no duplicates
+                mo_options.flags.extend(options["flags"])
+                mo_options.flags = list(set(mo_options.flags))
+
+            def patch_input_shape(deploy_cfg):
+                resize_cfg = get_configs_by_pairs(
+                    self._recipe_cfg.data.test.pipeline,
+                    dict(type="Resize"),
+                )
+                assert len(resize_cfg) == 1
+                resize_cfg = resize_cfg[0]
+                size = resize_cfg.size
+                if isinstance(size, int):
+                    size = (size, size)
+                assert all(isinstance(i, int) and i > 0 for i in size)
+                # default is static shape to prevent an unexpected error
+                # when converting to OpenVINO IR
+                deploy_cfg.backend_config.model_inputs = [ConfigDict(opt_shapes=ConfigDict(input=[1, 3, *size]))]
+
+            patch_input_preprocessing(deploy_cfg)
+            if not deploy_cfg.backend_config.get("model_inputs", []):
+                patch_input_shape(deploy_cfg)
+
+        return deploy_cfg
+
+    # This should be removed
+    def update_override_configurations(self, config):
+        """Update override_configs."""
+        logger.info(f"update override config with: {config}")
+        config = ConfigDict(**config)
+        self.override_configs.update(config)
+
+    def save_model(self, output_model: ModelEntity):
+        """Save best model weights in SegmentationTrainTask."""
+        logger.info("called save_model")
+        buffer = io.BytesIO()
+        hyperparams_str = ids_to_strings(cfg_helper.convert(self._hyperparams, dict, enum_to_str=True))
+        labels = {label.name: label.color.rgb_tuple for label in self._labels}
+        model_ckpt = torch.load(self._model_ckpt)
+        modelinfo = {
+            "model": model_ckpt,
+            "config": hyperparams_str,
+            "labels": labels,
+            "VERSION": 1,
+        }
+
+        torch.save(modelinfo, buffer)
+        output_model.set_data("weights.pth", buffer.getvalue())
+        output_model.set_data(
+            "label_schema.json",
+            label_schema_to_bytes(self._task_environment.label_schema),
+        )
+        output_model.precision = self._precision
+
+    # These need to be moved somewhere
+    def _update_caching_modules(self, data_cfg: Config) -> None:
```

Comment:
Do we have any consensus to move this module? looks like it's globally used as well.

Reply:
Same with above,
https://github.com/openvinotoolkit/training_extensions/blob/develop/otx/algorithms/detection/adapters/mmdet/task.py#L783
https://github.com/openvinotoolkit/training_extensions/blob/develop/otx/algorithms/classification/adapters/mmcls/task.py#L420

Reply:
This cache module is global but it use mmcv level configs. Currently we don't discuss about global mmTask, so there are many duplicated codes in mmTask such as configurer. Removing duplicated code in mmTask is also key content for next refactoring

Reply:
Thanks for explanation